W-15059250 AEH update troubleshooting for validate rule bypass (#44)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/troubleshooting.adoc

An error occurs when saving styling changes to the portal if an object has validation rules applied for standard objects that are created during the Salesforce linking process. 

Custom validation rules can be applied to several Salesforce objects, for example, account and user objects that cause linking and other processes to fail. 

The workaround is to bypass the validation rule for the Salesforce user who performs the linking process. 

NOTE: These instructions pertain to bypassing account object validation rules but can apply to bypassing other object custom validation rules.

. Create a custom permission:
.. From Salesforce, go to *Setup*, enter *custom permissions* in the Quick Find box, and select *Custom Permissions*.
.. Click *New* and enter the permission information:
+
* *Label*
+ 
Enter *Bypass Account Validation Rule*. 
+
* *Name* 
+
Click into the field to populate it with the label name with underscores.
+
* *Description*
+ 
Enter a description for the permission.
.. Click *Save*.

> . Create a permission set and add the custom permission:

Reviewer: This workaround is documented creating a new permission set, however will also possible add the custom permission to an existing permissions set like the one created for minimal permissions initiative. 
The one that is documented [here](https://docs.mulesoft.com/api-experience-hub/assigning-permissions).

Author: Thanks. I added that you can also use an existing permission set.

An error occurs when saving styling changes to the portal if an object has validation rules applied for standard objects that are created during the Salesforce linking process. 

> Validation rules cannot be applied to Salesforce standard objects, for example, account and user objects, that are created during the Salesforce linking process. 

Reviewer 2: ```suggestion
You can't apply validation rules to Salesforce standard objects, such as account and user objects, that are created during the Salesforce linking process. 
```

The workaround is to bypass the validation rule for the Salesforce user who performs the linking process. 

NOTE: These instructions pertain to bypassing account object validation rules but can apply to bypassing custom validation rules of other objects. 

. Create a custom permission:
.. From Salesforce, go to *Setup*, enter *custom permissions* in the *Quick Find* box, and select *Custom Permissions*.
.. Click *New* and enter the permission information:
+
* *Label*
+ 
Enter *Bypass Account Validation Rule*. 
+
* *Name* 
+
Click the field to populate it with the label name with underscores.
+
* *Description*
+ 
Enter a description for the permission.
.. Click *Save*.

> . Create a permission set and add the custom permission:

Reviewer 2: It's a little weird to have the colon introducing the sub-list broken up by the hanging paragraph. Is there another place you can put the following sentence?

+ 
You can also use an existing permission set instead of creating a new one.

.. From *Setup*, enter *permission sets* in the *Quick Find* box and select *Permission Sets*.
.. Click *New* and enter the permission set information:
+
* *Label*
+ 
Enter *Bypass Account Validation Rule*. 
+
* *Name* 
+
Click the field to populate it with the label name with underscores.
.. Click *Save*.
.. Click *Bypass Account Validation Rule* and select *Custom Permissions* from the *Apps* section.
.. From *Available Custom Permissions*, select *Bypass Account Validation*, click *Add*, and then click *Save*.
.. From the *Custom Permissions* list, select *Bypass Account Validation* and *Save*.
.. From *Object Manager*, select *Validation Rules*.
.. Select the object used for the validation rule and click *Edit*.
.. In the *Error Condition Formula* box, enter the following:
+
[codeblock]
----
AND (
  NOT($Permission.Bypass_Account_Validation),
  ISBLANK (Website)  
)
----
.. Click *Save*.

. Assign the permission set to the user performing the linking process:
+

> This user must have the System Administrator profile to perform the linking process.

Reviewer 2: (same as line 147)

Reviewer 2: Maybe move it to below line 184?
